http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcSecurityRealm.java

```diff
@@ -118,7 +118,11 @@ public SupportLevel getEvidenceVerifySupport(Class<? extends Evidence> evidenceT
             }
             RealmAccessClaim realmAccessClaim = accessToken.getRealmAccessClaim();
             if (realmAccessClaim != null) {
-                roles = realmAccessClaim.getRoles().stream().collect(Collectors.toSet());
+                if (roles == null) {
+                    roles = new HashSet<>(realmAccessClaim.getRoles());
+                } else {
+                    roles.addAll(realmAccessClaim.getRoles());
+                }
```

Comment:
Thanks for working on this @reinhapa! To simplify this a bit, we could initialize roles to `new HashSet<>()` as Viktar mentioned and then just use `roles.addAll` here. We can then remove the final `if` block at the bottom that checks for null.

Reply:
Done that, also moved out 3 calls to `session.getOidcClientConfiguration()` into a local variable